src/Components/App/App.js

```diff
@@ -34,46 +33,72 @@ class App extends Component {
 
   handleFavourite = (movie) => {
     let { currentUser, errorMsg, hasError } = this.props;
+    console.log(currentUser.favorites)
     if (currentUser.isSignedIn === true) {
-      addFavourite(movie, currentUser.id);
+      this.toggleFavourites(movie)
       hasError('');
     } else {
       hasError('Must be signed in to favourite!');
     }
   }
 
-  removeFavourite = async (id) => {
-    console.log('innit remove', id)
-    const { favouritesList } = this.props; 
-    console.log('THIS IS IN REMOVE', favouritesList.favorites);
-    let currentMovie = favouritesList.favorites.find(movie => movie.movie_id === id);
-     console.log(currentMovie);
-    //  let resp = await deleteFavorite(currentMovie.user_id, currentMovie.movie_id);
+  toggleFavourites = (movie) => {
+    console.log('movie', movie)
+    let { currentUser, favouritesList } = this.props;
+    console.log('og favouritesList', favouritesList)
+    let currentMovie = favouritesList.favorites.find(film => film.movie_id === movie.movie_id);
+    console.log(currentMovie)
+    if (currentMovie === undefined) { 
+      console.log('inside') 
+      addFavourite(movie, currentUser.id);
+      this.displayFavourites(currentUser.id);
+    } else {
+      console.log('else')
+      this.removeFavourite(currentMovie);
+      this.displayFavourites(currentUser.id);
+      console.log('removeFav', favouritesList)
+    }
+  }
 
+  removeFavourite = async (movie) => {
+    const { favouritesList, currentUser } = this.props; 
+    console.log('before', favouritesList)
+    // let currentMovie = favouritesList.favorites.find(movie => movie.movie_id === id);
      try {
-      isLoading(true);
-      const deletedmovies = await deleteFavorite(currentMovie.user_id, currentMovie.movie_id);
-      isLoading(false)
-      console.log(deletedmovies)
-      // getMovies(movies);
+      const deletedmovies = await deleteFavorite(movie.user_id, movie.movie_id);
+      this.displayFavourites(currentUser.id)
     } catch (error) {
       hasError(error.message)
     }
   }
 
+  displayFavourites = async (id) => {
```

Comment:
Great! We so needed to POST again to be able to get the updated favouritesList @ZoeKHarvey @SamuelColeman Yall are great!